src/toil/jobStores/azureJobStore.py

```diff
@@ -235,16 +236,72 @@ def _jobStoreExists(self):
         for attempt in retry_azure():
             with attempt:
                 try:
-                    table = self.tableService.query_tables(table_name=self.qualify('statsFileIDs'))
+                    table = self.locator.tableService.query_tables(self.qualify('statsFileIDs'))
                     return table is not None
                 except AzureMissingResourceHttpError as e:
                     if e.status_code == 404:
                         return False
                     else:
                         raise
 
+    def deleteJobStore(self):
+        self.cleanJobStore(self.locator)
+
+    @classmethod
+    def cleanJobStore(cls, locator):
+        i = 0
+        existed = False
+        failure = False
+        for name in ('files', 'statsfiles'):
```

Comment:
Use the `set` approach I talked about yesterday. Eliminate the while True loop and log_component_failure.